nslsii/iocs/epics_motor_ioc_sim.py

```diff
@@ -0,0 +1,161 @@
+                                      name='.FOFF')
+    set_use_switch = pvproperty(value=_false_true_states[0],
+                                enum_strings=_false_true_states,
+                                dtype=ChannelType.ENUM,
+                                name='.SET')
+
+    # configuration
+
+    _velocity = 1.
+    _acceleration = 3.
+
+    velocity = pvproperty(value=_velocity, read_only=True,
+                          dtype=ChannelType.DOUBLE,
+                          name='.VELO')
+    acceleration = pvproperty(value=_acceleration, read_only=True,
+                              dtype=ChannelType.DOUBLE,
+                              name='.ACCL')
+    motor_egu = pvproperty(value=_egu, read_only=True,
+                           dtype=ChannelType.STRING,
+                           name='.EGU')
+
+    # motor status
+
+    motor_is_moving = pvproperty(value='False', read_only=True,
+                                 enum_strings=_false_true_states,
+                                 dtype=ChannelType.ENUM,
+                                 name='.MOVN')
+    motor_done_move = pvproperty(value='False', read_only=False,
+                                 enum_strings=_false_true_states,
+                                 dtype=ChannelType.ENUM,
+                                 name='.DMOV')
+
+    high_limit_switch = pvproperty(value=0, read_only=True,
+                                   dtype=ChannelType.INT,
+                                   name='.HLS')
+    low_limit_switch = pvproperty(value=0, read_only=True,
+                                  dtype=ChannelType.INT,
+                                  name='.LLS')
+
+    direction_of_travel = pvproperty(value=_dir_states[1],
+                                     enum_strings=_dir_states,
+                                     dtype=ChannelType.ENUM,
+                                     name='.TDIR')
+
+    # commands
+
+    _cmd_states = ['False', 'True']
+
+    motor_stop = pvproperty(value=_cmd_states[0],
+                            enum_strings=_cmd_states,
+                            dtype=ChannelType.ENUM,
+                            name='.STOP')
+    home_forward = pvproperty(value=_cmd_states[0],
+                              enum_strings=_cmd_states,
+                              dtype=ChannelType.ENUM,
+                              name='.HOMF')
+    home_reverse = pvproperty(value=_cmd_states[0],
+                              enum_strings=_cmd_states,
+                              dtype=ChannelType.ENUM,
+                              name='.HOMR')
+
+    # Methods
+
+    @user_setpoint.putter
+    async def user_setpoint(self, instance, value):
+        p0 = instance.value
+        dwell = self._step_size/self._velocity
+        N = max(1, int((value - p0) / self._step_size))
+
+        await self.motor_done_move.write(value='False')
+
+        for j in range(N):
+            new_value = p0 + self._step_size*(j+1)
+            time.sleep(dwell)
```

Comment:
Please do not use sychronous sleeps in async functions, it will block the server while it is dwelling.